trio/_core/_tests/test_run.py

```diff
@@ -2025,15 +2078,18 @@ async def test_Task_custom_sleep_data():
 
 
 @types.coroutine
-def async_yield(value):
+def async_yield(value: T) -> Generator[T, None, None]:
     yield value
 
 
-async def test_permanently_detach_coroutine_object():
-    task = None
-    pdco_outcome = None
+async def test_permanently_detach_coroutine_object() -> None:
+    task: _core.Task | None = None
+    pdco_outcome: outcome.Outcome[str] | None = None
 
-    async def detachable_coroutine(task_outcome, yield_value):
+    async def detachable_coroutine(
+        task_outcome: outcome.Outcome[Any],
```

Comment:
I *think* this can just be `task_outcome[None]`...? But honestly I'm a bit confused by this test...